### PR TITLE
fixes mob/living mind gc issues

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Fixes mind GC issues with mob/living
this ought to go for a TM simply due to the amount of things that this COULD fuck up, looked good on my end but you never know
curren't

## Why It's Good For The Game
performance

## Testing
compiled, ran, mass genocide 
## Changelog
NPFC

### CODE REVIEW

**Review by S34NW — APPROVED** [review]

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -48,6 +48,8 @@
 				qdel(S)
 			else
 				S.be_replaced()
+	if(mind)
+		mind.current = null
```
Comment (farie82) [inline code/modules/mob/living/living.dm:52]: Is there a case where `current` can be something different? Like when ghosting or so?

Comment (GDNgit) [reply code/modules/mob/living/living.dm:52]: That would be mob/dead/observer, not a living type

Comment (farie82) [reply code/modules/mob/living/living.dm:52]: Aye, but will the original mob still have the mind attached to it? If that's the case then you'll unset the `current` while it's not  even the mob that's being deleted.

Comment (GDNgit) [reply code/modules/mob/living/living.dm:52]: You can re-enter your body multiple times, so yes

Comment (SteelSlayer) [reply code/modules/mob/living/living.dm:52]: How about `if(mind.current == src)` instead?

Comment (GDNgit) [reply code/modules/mob/living/living.dm:52]: Maybe? I'll try it tomorrow but I don't think it will have any actual effect on the implementation

Comment (SteelSlayer) [reply code/modules/mob/living/living.dm:52]: Yeah shouldn't. Just trying to address the "is their `mind.current` actually this src mob?" concern.

Comment (GDNgit) [reply code/modules/mob/living/living.dm:52]: > Yeah shouldn't. Just trying to address the "is their `mind.current` actually this src mob?" concern.

If that's not the case something horrible has gone wrong and it should be logged in some way then, I can change it tomorrow and add a stacktrace if it goes to shit

Comment (Fox-McCloud) [reply code/modules/mob/living/living.dm:52]:
```
	if(mind?.current == src)
		mind.current = null
```

Comment (GDNgit) [reply code/modules/mob/living/living.dm:52]: Applied, sorry this took awhile, just got home

**Review by Fox-McCloud — APPROVED** [review]

**Pull request merged**